File: CAPSI.Sante.Domain/Models/SQLserver/Patient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CAPSI.Sante.Domain.Models.SQLserver
{
    public class Patient
    {
        public Guid Id { get; set; }
        public string NumeroAssuranceMaladie { get; set; }
        public string Nom { get; set; }
        public string Prenom { get; set; }
        public DateTime DateNaissance { get; set; }
        public string Sexe { get; set; }
        public string Telephone { get; set; }
        public string Email { get; set; }
        public string Adresse { get; set; }
        public string CodePostal { get; set; }
        public string Ville { get; set; }
        public string GroupeSanguin { get; set; }

        // Nouvelle propriété pour la photo
        public string PhotoUrl { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public Guid? UserId { get; set; }
        public bool EstActif { get; set; } = true;

        // Navigation property
        public DossierMedical DossierMedical { get; set; }

        // Propriétés calculées pour l'interface utilisateur
        public string NomComplet => $"{Prenom} {Nom}";
        public int Age => DateTime.Today.Year - DateNaissance.Year -
                         (DateTime.Today.DayOfYear < DateNaissance.DayOfYear ? 1 : 0);
        public bool APhoto => !string.IsNullOrEmpty(PhotoUrl);
    }
}

File: capsi-sante-webapp/src/models/Patient.ts
export interface Patient {
  id: string; // Guid
  numeroAssuranceMaladie: string;
  nom: string;
  prenom: string;
  dateNaissance: string; // DateTime (ISO 8601 string)
  sexe: string; // M ou F
  telephone: string;
  email: string;
  adresse: string;
  codePostal: string;
  ville: string;
  groupeSanguin: string; // (A|B|AB|O)[+-]
  photoUrl?: string; // Nullable
  createdAt: string; // DateTime (ISO 8601 string)
  updatedAt?: string; // Nullable DateTime (ISO 8601 string)
  userId?: string; // Nullable Guid
  estActif: boolean;
}
